Modify GH workflow to always publish to latest

micrictor opened the pull request:
I ran into some dependency issues recently due to the `latest` image not reflecting what's in this repo. This change will make it so every commit to this repo generates a new `latest` image, while still allowing for the automatic creation of specifically tagged images.

I think that this is better all around - less work for the single-threaded owner of this repo, as `latest` is always up-to-date, consumers of the `serverless-rust` package automagically get the latest improvements to the build image, and users are still free to select specific versions as needed.

The only thing I'm unsure of is if, given this change, we would then want to set specific versions of the build image for versions of `serverless-rust`.

hrishikeshparanjape commented:
this one should solve https://github.com/softprops/lambda-rust/issues/98 I think.

zamazan4ik committed:
feat: initial changes after the fork

- update README with proper links
- refactor a little bit ignore files
- update changelog (add missing newer Rust versions)
- update Makefile defaults
- add me to the license file

Tested:
- Local tests

zamazan4ik committed:
doc: update README

- fix wrong links to old softprops repos

micrictor committed:
Fix test image reference

micrictor committed:
Rework GH actions to CD latest, tag on release

Commit:
Merge pull request #2 from micrictor/master

- fix test image reference

jerusdp committed:
Makefile updated to use Rust 1.55.0 as the default version

jerusdp committed:
Update change log for Rust version 1.55.0

Commit:
merge: Merge pull request #3 from jerusdp/rust-1.55.0

- Makefile updated to use Rust 1.55.0 as the default version

Tested:
- Local tests

Commit:
merge: Merge pull request #4 from jerusdp/master

- update change log for Rust version 1.55.0

zamazan4ik committed:
fix: fix publishing releases

- add new parameter to Makefile
- delete wrong 'with' clause from Github Actions file

Tested:
- No

jerusdp committed:
Add .vscode to ignored files

jerusdp committed:
Script to check cersion
Problem
Rust releases a new version approximatly every six weeks.
We need to detect a new release
Solution
* install the current stable toolchain
* extract the version of the stable toolchain from rustup check
* extract the version of the installed toolchain from rustup default
* compare and return an error if they do not match
Note
Docker containers are built with  a specific version (not stable).

jerusdp committed:
Check the latest R build
Problem
Script to check must be installed in the build
Solution
* Add the latest.sh script to container
Note

jerusdp committed:
Check the Rust version
Problem
Exceute the latest.sh script
Solution
* make command to run the latest.sh script entrypoint
Note

jerusdp committed:
Schedule Github Actions Version Check
Problem
Version check should be executed routinely
Solution
* Use scheduled github action to run check
* Load unix and checkout source
* Make check to run script
Note
* docker on host will download latest conainer

jerusdp committed:
FIX: On speficication

jerusdp committed:
Update check.yml

FIX: missing colon on cron

jerusdp committed:
Merge pull request #1 from jerusdp/latest

Latest

jerusdp committed:
Test using my docker

jerusdp committed:
Scheduled test for top of next hour

jerusdp committed:
manual trigger option

jerusdp committed:
Update check.yml

FIX: yaml error

jerusdp committed:
Create an issue

jerusdp committed:
Merge branch 'master' of github.com:jerusdp/lambda-rust

jerusdp committed:
Update check.yml

jerusdp committed:
Update check.yml

jerusdp committed:
Update check.yml

jerusdp committed:
Update check.yml

jerusdp committed:
Update check.yml

added checkout

jerusdp committed:
Update check.yml

checkout version2

jerusdp committed:
Update check.yml

FIX: ID created new step

jerusdp committed:
Update check.yml

Use different create issue action

jerusdp committed:
Update check.yml

jerusdp committed:
FIX: Update to write issue on failure

jerusdp committed:
Subject
Problem
Solution
Note

jerusdp committed:
FIX  - too much copying :(

jerusdp committed:
echo stable

jerusdp committed:
attempt to use gh to create the issue within the shell script
]

jerusdp committed:
Wrong variable being transferred

jerusdp committed:
Use gh instead of action

jerusdp committed:
Use gh instead of action

jerusdp committed:
FIX missing step indicator

jerusdp committed:
FIXadd checkout to get auth?

jerusdp committed:
FIX: Add environment with GITHUB_TOKEN

jerusdp committed:
FIX: Use gh to clone instead of checkout

jerusdp committed:
FIX Use checkout

jerusdp committed:
Add label to issue

jerusdp committed:
Change label to existing lable

jerusdp committed:
Create issue on the repository without checking out by using -R flag

jerusdp committed:
FIX spelling of workflow_dispatch

jerusdp committed:
Merge branch 'master' into latest

jerusdp committed:
FIX wayward space in -R

jerusdp committed:
Merge branch 'latest' of github.com:jerusdp/lambda-rust into latest

jerusdp committed:
Update cron schedule to run early Tuesday morning

jerusdp committed:
Restore Makefile after testing

jerusdp committed:
Update environent variable setting

jerusdp committed:
Remove original version of kernel-devel as its not updated by yum and instead a second version is installed

jerusdp committed:
Update main to include security scan and address issue #6

jerusdp committed:
Use github.repository to set REPO

jerusdp committed:
Conditional execution of publish on the correct public repository

jerusdp committed:
Add .vscode to gitignore and delete seurity.yml working file

jerusdp committed:
FIX: Correct the name of the master repository to rust-serverless/lambda-rust

jerusdp committed:
Added workflow to build nightly Rust each night at 0200

jerusdp committed:
Nigthly make script to build and push nightly Rust

jerusdp committed:
Set IMAGE based on enirvonment or default

jerusdp committed:
Add Tag to Makefile
Problem
Makefile only makes latest and tags with Rust version
Solution
- variable TAG to set the tag with default latest
- tag the debug build to allows debugging non-latest builds
Note

jerusdp committed:
Build and Publish Nightly Rust
Problem
Nighlty Rust is updated every night!
Solution
 - Action runs each night at 0200 UTC
 - Action can be dispatched from console
 - Run test job and make with environment set to use and tag nigthly
 - Run publsh job on successful test with nightly
 - Only run publish on rust-serverless repo
 - Publish issue if publication fails
Note
Not configured to run security scan (part of seperate PR).

jerusdp committed:
FIX: double quotes to single quotes

jerusdp committed:
FIX: test needs to use the nigthly container image

jerusdp committed:
FIX: need to build before test

jerusdp committed:
Added newline to nightly.yml

Commit:
merge: Merge pull request #9 from jerusdp/nightly

- Build and Publish Nightly Rust image

jerusdp committed:
Added missing newlines

jerusdp committed:
Added missing newlines

jerusdp committed:
Resolve conflict with Makefile updated by nightly

jerusdp committed:
Merge branch 'master' into latest

jerusdp committed:
Added missing newlines

Commit:
merge: Merge pull request #7 from jerusdp/secruity

- Secruity Scan for Docker Image and fix for Issue #6

Commit:
merge: Merge pull request #8 from jerusdp/latest

- add GitHub action to check Rust stable version against Rust installed in latest docker container

micrictor committed:
Add autobuild to stable
